### PR TITLE
chore: feature-gate `js-tracer` to get rid of `boa`

### PR DESCRIPTION
**Pull request by storopoli**
This PR makes the hardcoded `js-tracer` feature in the codebase optional.
This feature brings `boa` which is a JavaScript engine implemented in Rust.
It's used to provide a kind of console for basic CLI-based scripting like `geth` does.

I don't think this should be default but an optional feature since `boa` brings a LOT of dependencies and huge compile times.
Also some developers might just use `reth` without the console CLI-based stuff.

PS: sorry for the empty description draft PR. I was planning to do a tidy up PR but somehow I got caught up with stuff on Fri evening and the result was a WIP draft PR with no description.

### CODE REVIEW

File: crates/rpc/rpc-eth-api/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ workspace = true
 [dependencies]
 # reth
 revm.workspace = true
-revm-inspectors = { workspace = true, features = ["js-tracer"] }
+revm-inspectors = { workspace = true }
```
Comment (mattsse) [inline crates/rpc/rpc-eth-api/Cargo.toml:17]:
```suggestion
revm-inspectors.workspace = true
```

File: crates/rpc/rpc/Cargo.toml
Comment (mattsse) [inline crates/rpc/rpc/Cargo.toml:1]: not sure we can easily do this here because js tracing is required by the debug impl

**Review by mattsse — CHANGES_REQUESTED** [review]
we can remove the default feature from rpc-eth-types in this pr, and if feature gating js tracer in the rpc crate is necessary, we can also do that

**Comment by storopoli** [comment]
I will work on this and will update this PR.
Right now I am trying to see if I can feature-gate somehow the `debug` `impl`s in the `rpc` crate.

File: crates/rpc/rpc-builder/Cargo.toml [hunk 2]
```diff
@@ -66,3 +66,6 @@ reth-node-api.workspace = true
 tokio = { workspace = true, features = ["rt", "rt-multi-thread"] }
 serde_json.workspace = true
 clap = { workspace = true, features = ["derive"] }
+
+[features]
+reth-rpc-js-tracer = ["reth-rpc/js-tracer"]
```
Comment (Rjected) [inline crates/rpc/rpc-builder/Cargo.toml:71]: can we just call this `js-tracer`

File: crates/rpc/rpc-eth-api/Cargo.toml [hunk 3]
```diff
@@ -55,4 +55,6 @@ optimism = [
     "reth-primitives/optimism",
     "revm/optimism",
     "reth-provider/optimism",
-]
\ No newline at end of file
+]
+inspectors-js-tracer = ["revm-inspectors/js-tracer"]
+rpc-eth-types-js-tracer = ["reth-rpc-eth-types/js-tracer"]
```
Comment (Rjected) [inline crates/rpc/rpc-eth-api/Cargo.toml:60]: should probably also be just `js-tracer`

File: crates/rpc/rpc-server-types/src/module.rs [hunk 4]
```diff
@@ -195,7 +195,7 @@ impl FromStr for RpcModuleSelection {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         if s.is_empty() {
-            return Ok(Self::Selection(Default::default()))
+            return Ok(Self::Selection(Default::default()));
```
Comment (Rjected) [inline crates/rpc/rpc-server-types/src/module.rs:198]:
```suggestion
            return Ok(Self::Selection(Default::default()))
```

File: crates/rpc/rpc-builder/src/lib.rs [hunk 5]
```diff
@@ -1685,7 +1691,7 @@ impl TransportRpcModules {
     /// Returns [Ok(false)] if no ipc transport is configured.
     pub fn merge_ipc(&mut self, other: impl Into<Methods>) -> Result<bool, RegisterMethodError> {
         if let Some(ref mut ipc) = self.ipc {
-            return ipc.merge(other.into()).map(|_| true)
+            return ipc.merge(other.into()).map(|_| true);
```
Comment (Rjected) [inline crates/rpc/rpc-builder/src/lib.rs:1694]:
```suggestion
            return ipc.merge(other.into()).map(|_| true)
```

File: crates/rpc/rpc-builder/src/lib.rs [hunk 6]
```diff
@@ -1673,7 +1679,7 @@ impl TransportRpcModules {
     /// Returns [Ok(false)] if no ws transport is configured.
     pub fn merge_ws(&mut self, other: impl Into<Methods>) -> Result<bool, RegisterMethodError> {
         if let Some(ref mut ws) = self.ws {
-            return ws.merge(other.into()).map(|_| true)
+            return ws.merge(other.into()).map(|_| true);
```
Comment (Rjected) [inline crates/rpc/rpc-builder/src/lib.rs:1682]:
```suggestion
            return ws.merge(other.into()).map(|_| true)
```

File: crates/rpc/rpc-builder/src/lib.rs [hunk 7]
```diff
@@ -1661,7 +1667,7 @@ impl TransportRpcModules {
     /// Returns [Ok(false)] if no http transport is configured.
     pub fn merge_http(&mut self, other: impl Into<Methods>) -> Result<bool, RegisterMethodError> {
         if let Some(ref mut http) = self.http {
-            return http.merge(other.into()).map(|_| true)
+            return http.merge(other.into()).map(|_| true);
```
Comment (Rjected) [inline crates/rpc/rpc-builder/src/lib.rs:1670]:
```suggestion
            return http.merge(other.into()).map(|_| true)
```

**Review by Rjected — CHANGES_REQUESTED** [review]

File: crates/rpc/rpc-builder/Cargo.toml (hunk 2 above)
Comment (storopoli) [reply crates/rpc/rpc-builder/Cargo.toml]: Yes, done in 3e5f678f8431096f36ec7185de038b7a72b2b0c4

File: crates/rpc/rpc-eth-api/Cargo.toml (hunk 3 above)
Comment (storopoli) [reply crates/rpc/rpc-eth-api/Cargo.toml]: Done in 3e5f678f8431096f36ec7185de038b7a72b2b0c4

File: crates/rpc/rpc-server-types/src/module.rs (hunk 4 above)
Comment (storopoli) [reply crates/rpc/rpc-server-types/src/module.rs]: Got rugpulled by rustfmt being trigger-happy.
Undone in 3e5f678f8431096f36ec7185de038b7a72b2b0c4

File: crates/rpc/rpc-builder/src/lib.rs (hunk 5 above)
Comment (storopoli) [reply crates/rpc/rpc-builder/src/lib.rs]: Got rugpulled by rustfmt being trigger-happy.
Undone in 3e5f678f8431096f36ec7185de038b7a72b2b0c4

File: crates/rpc/rpc-builder/src/lib.rs (hunk 7 above)
Comment (storopoli) [reply crates/rpc/rpc-builder/src/lib.rs]: Got rugpulled by rustfmt being trigger-happy.
Undone in 3e5f678f8431096f36ec7185de038b7a72b2b0c4

**Comment by storopoli** [comment]
I fix the clippy lints. I might need some guidance in the `crate-checks` and `book` CI failures.

File: crates/rpc/rpc-builder/src/lib.rs [hunk 8]
```diff
@@ -132,6 +132,7 @@
 )]
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![feature(stmt_expr_attributes)]
```
Comment (Rjected) [inline crates/rpc/rpc-builder/src/lib.rs:135]:
```suggestion
```

**Review by Rjected — CHANGES_REQUESTED** [review]

File: crates/rpc/rpc-builder/src/lib.rs (hunk 8 above)
Comment (storopoli) [reply crates/rpc/rpc-builder/src/lib.rs]: done in 0afc55670748e75ab55169e30b491ceac640bac8

**Review by Rjected — COMMENTED** [review]
looks like some test utils need changes to work with the new features

File: crates/rpc/rpc-builder/src/lib.rs (hunk 8 above)
Comment (storopoli) [reply crates/rpc/rpc-builder/src/lib.rs]: Tests are failing because we are missing this `nightly` feature

**Review by Rjected — CHANGES_REQUESTED** [review]
Test failures are due to using cfgs on expressions, which is not stable, so we need to change things to work with stable

**Review by mattsse — CHANGES_REQUESTED** [review]
this feature is now way too invasive and leaks into cli crate.

this should be solved differently, so that this doesn't have to be configured top to bottom

this can also be done in smaller steps, e.g. making it non default first

**Comment by storopoli** [comment]
Thanks @mattsse and @nkysg for the review and for trying a principled-approach fix 🚀

**Comment by mattsse** [comment]
this is no no longer default in the rpc crates and is enabled in rpc-builder

if you'd like to enable this even higher, like on the node level lmk

**Comment by storopoli** [comment]
Yeah I would love to enable this in the node level as well, e.g. in the `rpc-builder`.

**Pull request closed without merging**